lib/reek/spec/should_reek_of.rb

```diff
@@ -18,6 +18,7 @@ def initialize(smell_type_or_class,
         @smell_details = smell_details
         @configuration = configuration
         @configuration.load_values(smell_type => { Smells::SmellConfiguration::ENABLED_KEY => true })
+        @matching_smell_types = nil
```

Comment:
After thinking about it this:

>> I think initialising instance variables to nil is a strong hint that.....this shouldn't be an instance variable at all.

is a separate smell for which we'd need a separate detector. That is, if we all agree on that it is a smell 😆 
So please ignore my comment here, I'll create a separate issue for this.